sdk/core/core-rest-pipeline/src/policies/bearerTokenAuthenticationPolicy.ts

```diff
@@ -219,3 +317,74 @@ export function bearerTokenAuthenticationPolicy(
     },
   };
 }
+
+/**
+ *
+ * Interface to represent a parsed challenge.
+ *
+ * @internal
+ */
+interface AuthChallenge {
+  scheme: string;
+  params: Record<string, string>;
+}
+
+/**
+ * Converts: `Bearer a="b", c="d", Pop e="f", g="h"`.
+ * Into: `[ { scheme: 'Bearer', params: { a: 'b', c: 'd' } }, { scheme: 'Pop', params: { e: 'f', g: 'h' } } ]`.
+ *
+ * @internal
+ */
+export function parseChallenges(challenges: string): AuthChallenge[] {
+  // Split the input string by schemes and parameters
+  // Challenge regex seperates the string to individual challenges with different schemes in the format `Scheme a="b", c="d"`
+  const challengeRegex = /(\w+) ((?:\w+="[^"]*",?\s*)+)/g;
+  // Parameter regex captures the claims group removed from the scheme in the format `a="b"` and `c="d"`
+  const paramRegex = /(\w+)="([^"]*)"/g;
+
+  const parsedChallenges: AuthChallenge[] = [];
+  let match;
+
+  // Iterate over each challenge match
+  while ((match = challengeRegex.exec(challenges)) !== null) {
+    const scheme = match[1];
+    const paramsString = match[2];
+    const params: Record<string, string> = {};
+    let paramMatch;
+
+    // Iterate over each parameter match
+    while ((paramMatch = paramRegex.exec(paramsString)) !== null) {
+      params[paramMatch[1]] = paramMatch[2];
+    }
+
+    parsedChallenges.push({ scheme, params });
+  }
+  return parsedChallenges;
+}
+
+/**
+ * Parse a pipeline response and look for a CAE challenge with "Bearer" scheme
+ * Return the value in the header without parsing the challenge
+ * @internal
+ */
+function getCaeChallengeClaims(response: PipelineResponse): string | undefined {
+  const challenges = response.headers.get("WWW-Authenticate");
+  // Find all challenges present in the header
+  const parsedChallenges = parseChallenges(challenges as string);
+  let caeChallenge: AuthChallenge | undefined;
+  // Find the CAE challenge
+  for (const challenge of parsedChallenges) {
+    if (
+      challenge.scheme === "Bearer" &&
+      challenge.params["claims"] &&
+      challenge.params["error"] === "insufficient_claims"
+    ) {
+      caeChallenge = challenge;
+    }
+  }
+  // Return empty when no CAE challenge is found
+  if (!caeChallenge) {
+    return;
+  }
+  return caeChallenge.params["claims"];
```

Comment:
nit: consider simplifying, something along these lines should work

```suggestion
  return parsedChallenges.find(x => x.sceme === "Bearer" && x.params.claims && x.params.error)?.claims;
```